tools/java/java-build/src/com/google/i18n/phonenumbers/BuildMetadataProtoFromXml.java

```diff
@@ -159,23 +160,36 @@ public boolean start() {
           BuildMetadataFromXml.buildPhoneMetadataCollection(inputFile, liteBuild);
 
       if (singleFile) {
+        int SINGLE_FILE_BUFFER_SIZE = 256 * 1024;
         FileOutputStream output = new FileOutputStream(filePrefix);
         ObjectOutputStream out = new ObjectOutputStream(output);
-        metadataCollection.writeExternal(out);
+        byte[] outputArray = new byte[SINGLE_FILE_BUFFER_SIZE];
+        CodedOutputByteBufferNano outputByteBuffer =
+            CodedOutputByteBufferNano.newInstance(outputArray);
+        metadataCollection.writeTo(outputByteBuffer);
```

Comment:
Should have a helper function that takes care of the CodedOutputByteBufferNano; e.g.:
```
writePhoneMetadataCollection(PhoneMetadataCollection, ObjectOutputStream);
```